pkg/cli/config/adminkubeconfig/new_admin_kubeconfig.go

```diff
@@ -0,0 +1,263 @@
+}
+
+var (
+	newAdminKubeconfigLong = templates.LongDesc(i18n.T(`
+		Generate, make the server trust, and display a new admin.kubeconfig.
+
+		The key is produced locally and never hits disk.  The public half is pushed to the cluster
+		for the kube-apiserver to trust the locally created admin.kubeconfig.`))
+
+	newAdminKubeconfigExample = templates.Examples(`
+		# Generate a new admin kubeconfig
+		oc config new-admin-kubeconfig`)
+)
+
+func NewAdminKubeconfigOptions(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *NewAdminKubeConfigOptions {
+	return &NewAdminKubeConfigOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+	}
+}
+
+func NewCmdConfigRefreshCABundle(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewAdminKubeconfigOptions(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "new-admin-kubeconfig",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Generate, make the server trust, and display a new admin.kubeconfig."),
+		Long:                  newAdminKubeconfigLong,
+		Example:               newAdminKubeconfigExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete())
+			cmdutil.CheckErr(o.Validate(args))
+			r, err := o.ToRuntime()
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(r.Run(context.TODO()))
+
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *NewAdminKubeConfigOptions) AddFlags(cmd *cobra.Command) {
+}
+
+func (o *NewAdminKubeConfigOptions) Complete() error {
+
+	return nil
+}
+
+func (o NewAdminKubeConfigOptions) Validate(args []string) error {
+	if len(args) > 0 {
+		fmt.Errorf("no arguments allowed")
+	}
+
+	return nil
+}
+
+func (o *NewAdminKubeConfigOptions) ToRuntime() (*RefreshCABundleRuntime, error) {
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &RefreshCABundleRuntime{
+		KubeClient: kubeClient,
+		Host:       clientConfig.Host,
+		IOStreams:  o.IOStreams,
+	}
+
+	return ret, nil
+}
+
+type RefreshCABundleRuntime struct {
+	KubeClient kubernetes.Interface
+	Host       string
+
+	genericclioptions.IOStreams
+}
+
+func (r *RefreshCABundleRuntime) Run(ctx context.Context) error {
+	systemMasterClientCert, signerCertBytes, err := createNewAdminClientCert()
+	if err != nil {
+		return fmt.Errorf("unable to create new client certificate: %w", err)
+	}
+
+	serverCABundle, err := refreshcabundle.GetCABundleToTrustKubeAPIServer(ctx, r.KubeClient)
+	if err != nil {
+		return fmt.Errorf("unable to get the CA bundle from the cluster: %w", err)
+	}
+
+	// update the in-cluster configmap
+	existingConfigMap, err := r.KubeClient.CoreV1().ConfigMaps("openshift-config").Get(ctx, "admin-kubeconfig-client-ca", metav1.GetOptions{})
```

Comment:
A very minor style thing, I would hoist `admin-kubeconfig-client-ca` to a `const` at the top so it's easily found (and use it in multiple places to avoid typos).